### PR TITLE
Adjust version string composition

### PR DESCRIPTION
**Pull request by Vezzra**
This would be my take on the new version string composition. I came up with the following for the "build number" (which would replace the svn rev number in our old implementation) and the version string:
- Compile the "build number" as follows: <2-digit-year-number><2-digit-month-number><2-digit-day-number>.<commit-hash>
- Use "git show -s --format=%h HEAD" to get the commit hash. The output of "git describe" can vary and is therefore more difficult to parse.
- Fetch the date of the commit and make sure to convert it to UTC.
- If the build is based on a branch other than master, set the branch name between the major version "0.4.4+" and the "build number".

So the final version string will be something like this: "v0.4.4+ branch [build 150320.831f1b0] Xcode 0511".

Dilvish has already made the suggestion on the forum to use 4-digit-year-number.

Further comments?

### CODE REVIEW

File: cmake/make_versioncpp.py [hunk 1]
```diff
@@ -49,8 +46,9 @@
...
 version_cpp.write(template.substitute(
     FreeOrion_VERSION=version,
-    FreeOrion_WC_REVISION=wc_rev,
+    FreeOrion_BRANCH=branch,
+    FreeOrion_BUILD_NO=build_no,
     FreeOrion_BUILDSYS=build_sys))
 version_cpp.close()
 
-print "Building rev: " + wc_rev
+print "Building v%s build %s" %(version, build_no)
```
Comment (genestack-solomatin) [inline cmake/make_versioncpp.py]: Git hub shows notification about no new line at the end of file. I ask Freeorion team no accept any pull request with this bug. (Click View full changes to see mark)

**Comment by genestack-solomatin** [comment]
Looks fine for me.
May be hide build postfix(Xcode 0511) if it is default?

File: cmake/make_versioncpp.py (hunk 1 above)
Comment (Vezzra) [inline cmake/make_versioncpp.py]: Added missing newline.

**Comment by Vezzra** [comment]
There is no default value for this postfix (build system: MSVC, Xcode or cmake).

**Pull request merged**